src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -53,34 +60,66 @@ def acr_task_create(cmd,  # pylint: disable=too-many-locals
                     resource_group_name=None,
                     target=None,
                     auth_mode=None):
+
+    registry, resource_group_name = get_registry_by_name(
+        cmd.cli_ctx, registry_name, resource_group_name)
+
+    if context_path.lower() == NULL_CONTEXT:
+        context_path = None
+        commit_trigger_enabled = False
+        pull_request_trigger_enabled = False
+
     if (commit_trigger_enabled or pull_request_trigger_enabled) and not git_access_token:
         raise CLIError("If source control trigger is enabled [--commit-trigger-enabled] or "
                        "[--pull-request-trigger-enabled] --git-access-token must be provided.")
 
-    if file.endswith(ALLOWED_TASK_FILE_TYPES):
-        FileTaskStep = cmd.get_models('FileTaskStep')
-        step = FileTaskStep(
-            task_file_path=file,
-            values_file_path=values,
+    yaml_template = ""
+    if cmd_value:
+        yaml_template = "steps: \n  - cmd: {{ .Values.command }}\n"
+        values_content = "command: {0}\n".format(cmd_value)
+
+    if file == "-":
+        import sys
+        for s in sys.stdin.readlines():
+            yaml_template += s
+        values_content = ""
+
+    if yaml_template and timeout:
```

Comment:
You need to move the timeout code into `if cmd_value`. 

    if cmd_value:
        yaml_template = "steps: \n  - cmd: {{ .Values.command }}\n"
        values_content = "command: {0}\n".format(cmd_value)
        if timeout:
            yaml_template += "    timeout: {}\n".format(timeout)

Reply:
What about timeout for interactive input, for instance, `az acr run --timeout 500 -f - /dev/null`?

Reply:
Which timeout will be honored? The timeout after --timeout or the timeout from the input stream?